src/Handlers/AbstractPaymentMethodHandler.php

```diff
@@ -568,23 +567,21 @@ private function preparePaymentsRequest(
                     $taxRate = 0;
                 }
 
+                $currency = $this->getCurrency(
+                    $transaction->getOrder()->getCurrencyId(),
+                    $salesChannelContext->getContext()
+                );
                 //Building open invoice line
                 $lineItems[] = $this->openInvoiceBuilder->buildOpenInvoiceLineItem(
                     $productName,
                     $this->currency->sanitize(
                         $price->getUnitPrice() -
```

Comment:
does the unit price already include the discount? 
I can imagine in case it's a percentage discount then yes but what if it's like 5 eur discount from the order total, is it still handled well in this case?

Reply:
No the price for each line item will not include the discount, normally the discount will be added as one of the line items with a negative value, e.g `"amountExcludingTax":"-120"`, and the total added up will be equal to the amount in `amount.value` of the request. But in this case, the discount is not included, and the total amount (adding up the line items) will be greater that the figure in `amount.value`.
```json
...
"amount": {
    "currency": "EUR",
    "value": 3000
  },
"lineItems": [
    {
      "id": "feac1274086a4531a9755492eb135102",
      "description": "Rustic Steel Aqua Radiant",
      "amountExcludingTax": 50467,
      "taxAmount": 3533,
      "taxPercentage": 700,
      "quantity": 1,
      "taxCategory": ""
    },
    {
      "id": "fe3b463d1aac4e1dbd4007b7dc47bb88",
      "description": "Aerodynamic Bronze iPlug",
      "amountExcludingTax": 88700,
      "taxAmount": 0,
      "taxPercentage": 0,
      "quantity": 1,
      "taxCategory": ""
    },
    {
      "id": "f2f6272cfb8d46d485020aa3cfca6712",
      "description": "Aerodynamic Cotton breezz",
      "amountExcludingTax": 56555,
      "taxAmount": 10745,
      "taxPercentage": 1900,
      "quantity": 1,
      "taxCategory": ""
    }
  ],
```
But the API will figure it out and add the discount line in the request to klarna

Reply:
let's create a follow up ticket for this because indeed the API will adjust the value but it's not the best approach to rely on only the API :(